Source/Core/Absy.cs

```diff
@@ -2871,6 +2892,33 @@ public Function(IToken tok, string name, List<TypeVariable> typeParams, List<Var
       DefinitionAxiom = new Axiom(tok, def);
       return DefinitionAxiom;
     }
+
+    // Generates function definition of the form func_call == func_body
+    // For example, for
+    // function {:define} foo(x:int) returns(int) { x + 1 }
+    // this will generate
+    // foo(x):int == x + 1
+    // We need the left hand call part later on to be able to generate
+    // the appropriate SMTlib style function definition.
+    public NAryExpr CreateFunctionDefinition(Expr body) {
```

Comment:
I find it odd to generate a DefBody in the SMTLIB style Absy.cs.  I was expecting that you would keep only the body expression here and then generate the function call etc on the left-hand side as needed when you are creating the actual SMT input.

Reply:
I agree. I invested quite a bit of time trying to make this cleaner. The tricky part is resolution, since you have to create the "function call" from function parameters that did not go through the resolution process, and that still have to be connected to the appropriate variables in the function body. At the point of VC generation, where function definition gets created, I could not figure out how to do this reliably.

Reply:
ok, understood.  Let us go ahead and land this PR.  We will investigate and improve gradually.